### PR TITLE
Support a --config file

### LINKED ISSUE #38: accept a config file
The config struct is already built out. As an additional option, we should allow users to pass a config file to the command line tool instead of specifying a bunch of args. It can be a JSON config file for now if that makes this issue easier. But I think YAML shouldn't be a huge pain to implement.

### PR DESCRIPTION
**Pull request by ncdc**
Only the last commit is relevant - will rebase after #103 is merged

Fixes #38

### CODE REVIEW

**Comment by ncdc** [comment]
Rebased

File: main.go [hunk 1]
```diff
@@ -22,12 +24,30 @@ func newLogger() logr.Logger {
...
+				}
+				defer reader.Close()
+
+				const bufferSize = 512
+				decoder := yaml.NewYAMLOrJSONDecoder(reader, bufferSize)
+				if err := decoder.Decode(&upgradeConfig); err != nil {
+					return errors.Wrap(err, "error parsing config file")
+				}
+
+				// TODO either merge in other flag values after decoding, or raise an error if the user specifies
```
Comment (chuckha) [inline main.go]: let's turn this into an issue so we can discuss approaches

Comment (chuckha) [reply main.go]: (i will do this)

**Comment by ncdc** [comment]
@chuckha I updated the readme!

**Pull request merged**